stock_push_delay/models/stock.py

```diff
@@ -0,0 +1,25 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class StockMove(models.Model):
+
+    _inherit = 'stock.move'
+
+    def _push_apply(self):
+        """Manual triggering"""
+        if self.env.context.get('manual_push', False):
+            return super(StockMove, self)._push_apply()
+
+
+class StockPicking(models.Model):
+
+    _inherit = 'stock.picking'
+
+    @api.multi
+    def action_done(self):
+        res = super(StockPicking, self).action_done()
+        for picking in self:
+            picking.move_lines.with_context(manual_push=True)._push_apply()
```

Comment:
Don't do that with context, so bad for performances.

Either you create another method (e.g.: _push_apply_manual()) , either you put context on self.

Reply:
Thanks for the comment, did not realize about that performance issue.